[Cognitive Services - Language] - Text CustomHealth - public preview

kayousef opened the pull request:
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

FumingZhang committed:
[Hub Generated] Publish private branch 'containerservice/official/v20221218' (#21987)

* [AutoSync] 6a6ad0e3336 Merged PR 7353698: [API Release] Sync swagger changes in GitHub PR #21069

* [AutoSync] c7b8955aefd Merged PR 7359461: [API Release] Fix Swagger LintDiff errors

* [AutoSync] 76fc4392c08 Merged PR 7379598: [Swagger][2022-11 APIs] azure auth-provider kubeconfig is fully removed in v1.26 instead of v1.25

Co-authored-by: swagger-automation <swagger@microsoft.com>

dsouzaarun committed:
Event Hub API version 2022-10-01-preview (#20724)

* Base Checkin

* Replace API version with 2022-10-01-preview

* Adding settings for Compaction

* Fixed examples

* Adding new api version to README.md

* Getting PolicyRetention enum to be modelled as string

* Adding latest api as default

* Fixing LintDiff

* Fixing Model Validation

* updated readme files

* Splitting the RetentionTimein Hours into two seperate properties.

* lint fix

* fixed typo

* Revert "fixed typo"

This reverts commit 2387ca3bf6f02e3a2df4966de2234c9dd2212e2b.

* Revert "lint fix"

This reverts commit 2c775034f650404cb9fe85bcc1730cb8425847b5.

* Revert "Splitting the RetentionTimein Hours into two seperate properties."

This reverts commit caf5d9ebaf2d5429ce489ba8004e805e5bd9159c.

* Fixing examples

* removed the supperssion

* added headers - azure-asyncoperation

* fixing the CI failures

* Revert "added headers - azure-asyncoperation"

This reverts commit 2d60d5c0d8d7839da3e6cf2d5f1aeba90c1efacb.

* updated CI fixes

* headers fix for cluster catch

Co-authored-by: REDMOND\ardsouza <ardsouza@microsoft.com>
Co-authored-by: davadhani <davadhani@microsoft.com>
Co-authored-by: Ajit Maruti Navasare (MINDTREE LIMITED) <v-ajnav@microsoft.com>
Co-authored-by: Ajit Navasare <ajitnavasare@microsoft.com>

sanjaiganesh committed:
Update Data container ubuntu os version (#21995)

* Add blockchain to latest profile

* Add additional types

* update

Co-authored-by: Mark Cowlishaw <markcowl@microsoft.com>

rishikaparashar committed:
[Hub Generated] Review request for Search to add version preview/2022-09-01-preview (#21898)

* Boundary Documentation changes in Dev maps search 2022 09 01 preview (#21892)

* Adds base for updating Search from version preview/2022-02-01-preview to version 2022-09-01-preview

* Updates readme

* Updates API version in new specs and examples

* Rename geocoding.json to search.json

* Added Boundary service APIs

* Fixed references to common spec

* Use an array instead of a map for geometries copyright

* Remove readOnly attribute from responses and default/max for the top parameter

* Mark seach/polygon location parameter as required

* Rename entityType parameter to resultType

* Rename entityType parameter to resultType in the example too

* Use View parameter from Common spec

* Doc Updates:1. Updated 'location' request parameter to 'coordinates' and updated the GetPolygon example with the correct values 2. Updated the boundary description to mention the expected geometry objects in the boundary API response 3. Updated View description to mention link to azure maps supported views details.

* Update custom-words.txt

Added POIs word to fix spell check error.

Co-authored-by: Igor Shishkin <specialforest@users.noreply.github.com>
Co-authored-by: Igor Shishkin <igshishk@microsoft.com>

* Update common.json (#21897)

* Updated the Boundary parameters description. (#21913)

* Updated the Boundary description and removed pois from custom words list.

* spelling validation fix for POIs - adding POIs to the custom-words list.

* Updated POIs description.

* addressed PR comment - Updated View description.

* Updated the Boundary view parameter description - default value set to auto.

Co-authored-by: Igor Shishkin <specialforest@users.noreply.github.com>
Co-authored-by: Igor Shishkin <igshishk@microsoft.com>

12VamshiKrishna committed:
EnableInteractivequery Changes (#21936)

* EnableInteractivequery Changes

* Spell Check

Co-authored-by: VamshiKrishna Chirra (Quadrant Resource) <vamshikrishna.chirra@quadrantresource.com>

chandrasekarendran committed:
Adding new api-version 2022-12-01 to Microsoft.DataProtection (#21661)

* Base commit for updating Microsoft.DataProtection from version stable/2022-05-01 to version stable/2022-12-01

Base commit for updating Microsoft.DataProtection from version stable/2022-05-01 to version stable/2022-12-01

* Updating readme

* Updating api version in new specs and examples

* Updating examples

* Adding CSR related changes

* Adding softDeleteSettings to backupVault

* Adding examples

* fixing swagger

* Fixing readme

* Adding description

* Adding immutability settings

* Update dataprotection.json

Added Expiry time field for RP contract

* Update ListRecoveryPoints.json

Updated ListRP example for expiry Time

* Merging latest fix

* Adding OperationalStore enum value

* Using parameters from v4 common

* Adding parameter for BackupInstanceName

* PatchResourceGaurd Fix

* Adding identity property in PatchResourceGuardInput

* Removing identity in resourceGuard resource

* Removing identity from ResourceGuard patch

Co-authored-by: amchandn <43201777+amchandn@users.noreply.github.com>
Co-authored-by: cvarsh <cvarsh@microsoft.com>
Co-authored-by: Srinivas Charan Madu <smadu@microsoft.com>

manikprabhu1208 committed:
s360 kpi fix (#22013)

msyyc committed:
Create readme.python.md (#21972)

* Create readme.python.md

* Update readme.md

bexxx committed:
Fix required api version parameter in OpenAI (#21921)

* Fix API version in swagger info for OpenAI

* Fix api version parameter of open ai ga version

navba-MSFT committed:
{AzureFrontdoor} fixes Azure/azure-cli#24758 fixing ActionType enum to include AnomalyScoring (#21947)

fixes Azure/azure-cli#24758 fixing ActionType enum to include AnomalyScoring

Running the below command to override the managed rules with --action set to AnomalyScoring fails:
> az network front-door waf-policy managed-rules override add

```
az network front-door waf-policy managed-rules override add: 'AnomalyScoring' is not a valid value for '--action'. Allowed values: Allow, Block, Log, Redirect.

```

This is because currently only the `Allow, Block, Log, Redirect` enums are specified. `AnomalyScoring` value should also be included in the `ActionType`  enums.

thramu committed:
Creating Settings.json spec under preview (#21723)

* Creating Settings.json spec under preview

* Adding settings.json into readme.md

* Prettier fix

* Review comments

Commit:
Feature/cplat 2022 11 01 (#21571)

* add 2022-11-01 folder

* update readme, update version reference

* Added userData property for PUT Restore Point API

Added user data property for PUT Restore Point API

* Revert "Added userData property for PUT Restore Point API"

This reverts commit 94c96d8ca8586b37a59083efd6357ccf2594dd61.

* Reimage Support for Non-Ephemeral VMs (#21090)

* Reimage support for non-ephemeral VMs

* small changes

* small changes

* resolving issues

* pretty check

* pretty install

* remove package-lock

* remove package-lock

* Added userData property in Restore Point PUT API endpoint (#21262)

* Added userData property in Restore Point PUT Operation

* Update based on review comment

* Updated description

Co-authored-by: Ashwani Tiwari <ashwtiwari@microsoft.com>

* Add OS Image profile to scheduled events profile and maxSurge in rolling upgrade policy in Microsoft.Compute (#21253)

* add reimage notification

* update reimage notification, add max surge

* fix grammar

* Adding requireGuestProvisionSignal and constrainedMaximumCapacity mis… (#21433)

* Adding requireGuestProvisionSignal and constrainedMaximumCapacity missing properties

* Moved the property at the end

Co-authored-by: Avaiyang Garg <avaiyanggarg@micrososft.com>

* Add serviceArtifactReference property (#21607)

* Add serviceArtifactReference property

* Moved the property inside the virtualMachineScaleSet file

* Minor change in example

Co-authored-by: Avaiyang Garg <avaiyanggarg@micrososft.com>

* add platformFaultDomain in instanceView (#21605)

* add platformFaultDomain in instanceView

* change platformFaultDomain values

* add APIVersionPattern Suppression (#21675)

Co-authored-by: Theodore Chang <theodore.l.chang@gmail.com>

* Adding imageDeprecationStatus properties to VirtualMachineImage response (#21615)

* Changes to virtualMachineImage.json to include ImageDeprecationStatus properties

* Changes to example json

* Fixed LintDiff errors

* Addressed review comment

* Introduce rollbackFailedInstancesOnPolicyBreach in VMSS RollingUpgradePolicy  (#21676)

* add property in compute.json

* add in 2022 file

* add rollbackFailedInstancesOnPolicyBreach in examples

* Revert "add property in compute.json"

This reverts commit 15a1e10503cc4e34b0b69f351cc8f23339002360.

* remove space

Co-authored-by: Sachin Doddaiah Shylaja <sadoddai@microsoft.com>
Co-authored-by: Theodore Chang <thchan@microsoft.com>

* fix spellingCheck error, LintDiff error, model validation, and typo

* fix modelValidation

* model validation

* update headers for examples

* model validation errors

* model validation

* Revert "model validation"

This reverts commit d3810a7866a9093b6ef75a57834e1aaeb4e139f3.

* Revert "model validation errors"

This reverts commit ba34d1b2af2bef58b60c16e9ed1cda00803a2ce0.

* Revert "update headers for examples"

This reverts commit 216b3dc1367655b8c7867a9d7a772fe7e2fa1583.

* Revert "model validation"

This reverts commit 285345656321e984bc99486a575e35a4d3359667.

* Revert "fix modelValidation"

This reverts commit a5fa156d42130566b827df88d6dea4ee591f4a6c.

* Revert "fix spellingCheck error, LintDiff error, model validation, and typo"

This reverts commit 9f60b71000be72bf4511feefcda0820b570345de.

* fix spelling error CI, file name typo, Lint Diff

Co-authored-by: Theodore Chang <theodore.l.chang@gmail.com>
Co-authored-by: Ashwani Tiwari <ashwtiwari@microsoft.com>
Co-authored-by: srcharug <62158390+srcharug@users.noreply.github.com>
Co-authored-by: csassasin <89677872+csassasin@users.noreply.github.com>
Co-authored-by: John Zhang <116316220+johnzhang-microsoft@users.noreply.github.com>
Co-authored-by: avaiyanggarg <116031789+avaiyanggarg@users.noreply.github.com>
Co-authored-by: Avaiyang Garg <avaiyanggarg@micrososft.com>
Co-authored-by: parahluwalia <117622623+parahluwalia@users.noreply.github.com>
Co-authored-by: RajalakshmiSwaminathan <raswamin@microsoft.com>
Co-authored-by: Sachin Doddaiah Shylaja <sachin.ds92@gmail.com>
Co-authored-by: Sachin Doddaiah Shylaja <sadoddai@microsoft.com>

XiaofeiCao committed:
readme.java.md (#22040)

leniatgh committed:
Check in Appconfig scenarios and recording assets (#21798)

* add appconfig control-plane scenarios

* touch swagger

* push tag successfully

* check in tag

* move assets.json to scenarios folder

* update tag

* add appconfig data-plane scenarios

* push data-plane apitest recording

* update mgmt recording

* restore swagger

* update recording

deyaaeldeen commented on dev/cognitiveservices/data-plane/Language/analyzetext.json:
```diff
@@ -831,6 +839,20 @@
+      },
+      "allOf": [
+        {
+          "$ref": "common.json#/definitions/PreBuiltTaskParameters"
```
Are we sure we want to use prebuilt params instead of [custom params](https://github.com/Azure/azure-rest-api-specs/blob/7eda52e0792c584c722016ef5ee533f923ba41cf/dev/cognitiveservices/data-plane/Language/analyzetext.json#L560)?

kayousef replied:
You're right. Thanks for pointing this out.

deyaaeldeen commented on dev/cognitiveservices/data-plane/Language/analyzetext.json:
```diff
@@ -864,6 +886,25 @@
+        "documents": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/HealthcareEntitiesDocumentResultWithDocumentDetectedLanguage"
```
Is the custom model guaranteed to assign the same categories the prebuilt model assigns?

kristapratico replied:
Good point, if so, then the custom healthcare task will need to accept task parameters like fhirVersion and documentType

kayousef replied:
You're right. Thanks for pointing this out.

deyaaeldeen replied:
I meant to ask about these [categories](https://github.com/Azure/azure-rest-api-specs/blob/6c192122507728c8ad9eb999e7970262f45673c3/specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzetext.json#L925), are custom health models guaranteed to detect entities of these categories? I understand that the list of categories is modeled as extensible enum but I want to make sure documented ones will also be detected.

kayousef replied:
@deyaaeldeen Yes, the custom API is guaranteed to return all prebuilt entities

deyaaeldeen commented on dev/cognitiveservices/data-plane/Language/analyzetext.json:
```diff
@@ -984,6 +1025,13 @@
             "$ref": "#/definitions/HealthcareEntityLink"
           }
+        },
+        "EntityComponentInformation": {
+          "description": "(Optional) Entity component information listing fired components of the extracted entity.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EntityComponentInformation"
+          }
```
I think the name of the model being used for each item should be different from the parent one.

deyaaeldeen replied:
Ah the capitalization threw me off and I thought it was a standalone definition, never mind this comment.

deyaaeldeen commented on dev/cognitiveservices/data-plane/Language/analyzetext.json:
```diff
@@ -984,6 +1025,13 @@
             "$ref": "#/definitions/HealthcareEntityLink"
           }
+        },
+        "EntityComponentInformation": {
```
Where is this array being used? Is the intention to add it to `HealthcareEntitiesDocumentResultWithDocumentDetectedLanguage`?

kristapratico commented on dev/cognitiveservices/data-plane/Language/analyzetext.json:
```diff
@@ -984,6 +1025,13 @@
             "$ref": "#/definitions/HealthcareEntityLink"
           }
+        },
+        "EntityComponentInformation": {
```
The name of the property should be camelCase

kristapratico replied:
yes

kayousef replied:
Why would it be camelCase? All other properties in the swagger are PascalCase (e.g. HealthcareEntityLink, HealthcareEntity, etc...)

kristapratico replied:
Definitions of types like HealthcareEntityLink, HealthcareEntity are PascalCase, but names of properties like confidenceScore should be camelCase.

kayousef replied:
Can you check again. The property name `entityComponentKind` in line 1195 is already camelCase

kristapratico replied:
I'm talking about camelCase for EntityComponentInformation which is listed as a property of HealthcareEntity:

https://github.com/Azure/azure-rest-api-specs/blob/10e48404583cb7c6d4bd98b1c461902ca36c32c3/dev/cognitiveservices/data-plane/Language/analyzetext.json#L1043-L1050

The definition (PascalCase) is on L1190: https://github.com/Azure/azure-rest-api-specs/blob/10e48404583cb7c6d4bd98b1c461902ca36c32c3/dev/cognitiveservices/data-plane/Language/analyzetext.json#L1190

kayousef replied:
I finally saw it! - Fixed.

kristapratico commented on dev/cognitiveservices/data-plane/Language/analyzetext.json:
```diff
@@ -1125,6 +1173,77 @@
+            "ListComponent"
+          ],
+          "x-ms-enum": {
+            "name": "componentKind",
```
Would the name `EntityCompenentKind` here work?

kayousef replied:
I think this would be redundant as the main object is called EntityComponetInformation anyway. 
I'm flexible here, but let's wait for the board feedback this week.

kayousef replied:
Okay, fair. I updated it

kristapratico commented on dev/cognitiveservices/data-plane/Language/analyzetext.json:
```diff
@@ -291,6 +297,7 @@
         "KeyPhraseExtraction",
         "EntityLinking",
         "Healthcare",
+        "CustomHealthcare",
```
I think we're missing the corresponding CustomHealthcareLROTask object that sets the discriminator. For example, the one for HealthcareLROTask: https://github.com/Azure/azure-rest-api-specs/blob/7eda52e0792c584c722016ef5ee533f923ba41cf/dev/cognitiveservices/data-plane/Language/analyzetext.json#L856-L869

kayousef replied:
Added. Thanks!

maddiemlucas committed:
change array name to settings (#21335)

thomasp98296 committed:
Task 16564293: Update swagger description of new budget alert. (#21879)

* Task 16564293: Update swagger description of new budget alert.

* Also remove "budget" as it will not longer be used.

* Add custom words

Co-authored-by: Thomas Pham <thompham@microsoft.com>

dhu5432 committed:
Add additionalProperties to Microsoft.ResourceGraph/resourcesHistory (#21088)

* Additional Properties

* Not anonymous

* PrettierCheck

Commit:
add or modify files (#21989)

navba-MSFT committed:
{AzureAPIM} fixes Azure/azure-rest-api-specs#21849  'in body' parameter specification missing (#21875)

* {AzureAPIM} fixes Azure/azure-rest-api-specs#21849  'in body' parameter specification missing

fixes Azure/azure-rest-api-specs#21849  'in body' parameter specification missing

The CreateOrUpdate operations on both of the following resource types are missing the in body parameter in the OpenAPI specification (OAS):

Microsoft.ApiManagement/service/contentTypes
Microsoft.ApiManagement/service/contentTypes/contentItems
This causes the documentation TryIt and autogenerated libraries (e.g. nodejs @azure/arm-apimanagement) to break, as they don't support passing in the properties in the body, making these resource types unable to be updated or created.

For example, if you run the "Try It" for this API, it doesn't give you a body field in the auto-generated REST HTTP request form. The body field is needed, but it doesn't give you that field as it's not populated in the OAS for this resource type.

See this: https://github.com/Azure/azure-rest-api-specs/blob/7245d0f5cc01bf95df9d7ae893f252d4001b4ad7/specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2021-08-01/apimcontenttypes.json#L132-L162

* Update apimcontenttypes.json

heaths commented on dev/cognitiveservices/data-plane/Language/analyzetext.json:
```diff
@@ -1125,6 +1173,77 @@
+          "description": "The component object kind.",
+          "type": "string",
+          "enum": [
+            "PrebuiltComponent",
+            "LearnedComponent",
+            "ListComponent"
```
In addition to what @kristapratico said about the name to better differentiate against something as common as "ComponentKind" (for a generate model name), enum values should also be camelCase. If PascalCase is ubiquitous, I would still ask that you verify this matches what the code does. Several times in Language, the enum values in the swagger and what's actually returned/accepted haven't aligned and it's been very problematic.

kayousef replied:
For componetKind, I feel adding "entity" is redundant. However, we can get the board feedback in this week's meeting and will follow the consensus.

For the enum casing, I'm following a similar structure in CLU [here ](https://github.com/Azure/azure-rest-api-specs/blob/main/dev/cognitiveservices/data-plane/Language/analyzeconversations.json#L1510) .. The enum is PascalCase, and I'm trying to align with existing APIs as much as possible.

Noted for the alignment with the service code. @mshaban-msft Please take a note of this. Let's make sure the service APIs respect the casing of the swagger enums.

heaths reviewed (changes requested):
I want to take another look when the issues @kristapratico and @deyaaeldeen raised are resolved.

Alancere committed:
[Go] add servicenetworking readme.go.md (#22082)

sameerpatro committed:
ASE New API version 2023-01-01-preview Changes (#22072)

* Base Commit from previous released version

* New API version changes

ArcturusZhang committed:
fix cosmosdb examples (#22087)

amperugu committed:
Add new tier and sku - Standard_Basic for Application Gateway (#21976)

* Add new tier and sku for standard_basic

* spaces

Commit:
Add scan folder and spark config support in Sparkjob activity (#21768)

* add scan folder and spark config support

* fix typo

* update description

* add custom word

kristapratico committed:
[Language - QA authoring] correct `source` to required param for update sources (#22065)

* correct source to required param for update sources

* fix validation errors while we're here

leniatgh committed:
Document how to integrate test-proxy (#21811)

* document how to integrate test-proxy

* update doc

* fix comments

guptashweta9 committed:
RAN Info Updated (#22101)

raych1 committed:
Adding cadl sample project (#21722)

* Adding cadl sample project

* cadl sample: packaqe.json & cadl-project.yaml (#21903)

* cadl sample: packaqe.json & cadl-project.yaml

* Update cadl-project.yaml

* Use faked shared model

* Update documentation/cadl-sample-project/specification/contosowidgetmanager/Contoso.WidgetManager/cadl-project.yaml

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

* Update cadl-project.yaml

* Adjusted to minimum file set

* update .Net emitter configuration

* update java emitter config

* Fixed examples issues

* Fixed example style issue

Co-authored-by: Wei Dong <40835867+dw511214992@users.noreply.github.com>
Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

abhinav-ghai committed:
Add 2023-02-27-preview for DigitalTwins (#21953)

* Add 2023-02-27-preview for DigitalTwins

* Update readme.md

raych1 committed:
Updated Toc in cadl structure markdown document (#22118)

IndraneelPal committed:
Changed Ehc Resource Id property to camel case (#22019)

deyaaeldeen commented on dev/cognitiveservices/data-plane/Language/analyzetext.json:
```diff
@@ -984,6 +1039,13 @@
             "$ref": "#/definitions/HealthcareEntityLink"
           }
+        },
+        "entityComponentInformation": {
+          "description": "(Optional) Entity component information listing fired components of the extracted entity. This object only applies for custom healthcare.",
```
Could this be improved by creating another definition that uses `allOf` original properties here and then add custom-specific properties like `entityComponentInformation` to the mix instead of adding it to a single shared definition?

@heaths What do you think?

deyaaeldeen commented on dev/cognitiveservices/data-plane/Language/analyzetext.json:
```diff
@@ -1125,6 +1187,77 @@
+      "type": "object",
+      "x-ms-discriminator-value": "PrebuiltComponent",
+      "properties": {
+        "value": {
```
how should this property be interpreted by customers?

deyaaeldeen reviewed (approved):
I would rather not do the API version bump here to simplify things but I don't feel strongly about it. I left a couple minor remarks but otherwise looks good to me. I did regenerate the JS SDK with it and it looks good.

ruowan commented:
/azp run

kristapratico commented on dev/cognitiveservices/data-plane/Language/examples/text/SuccessfulCustomHealthcareTaskStatusRequest.json:
```diff
@@ -0,0 +1,249 @@
+                    "warnings": []
+                  }
+                ],
+                "errors": []
```
example should also return projectName and deploymentName (should fix the ModelValidation error)

kayousef replied:
Fixed

bitsforthought committed:
Removing myself from CODEOWNERS since I no longer work on Azure Search (#22033)

Co-authored-by: Bruce Johnston <brjohnst@microsoft.com>

thatChang committed:
Release botservice microsoft.bot service 2022 09 15 (#22077)

* Base commit from 2022-06-15-preview

* Fix S360 issues

Co-authored-by: Chang Liu <liuchang@microsoft.com>

joseharriaga committed:
Add entry to CODEOWNERS (#22115)

msyyc committed:
update (#22144)

guptashweta9 committed:
Ran units Information updated (#22130)

* RAN Info Updated

* Ran Unit Info Updated

kayousef commented:
> I would rather not do the API version bump here to simplify things but I don't feel strongly about it. I left a couple minor remarks but otherwise looks good to me. I did regenerate the JS SDK with it and it looks good.

Definitely. Once the other PR with the version rename merges, I'll rebase this so it will only has custom health changes.

heaths reviewed (changes requested):
I like the changes, but had a couple of concerns regarding some minor changes.

heaths commented on dev/cognitiveservices/data-plane/Language/analyzetext.json:
```diff
@@ -1125,6 +1187,77 @@
+      "type": "object",
+      "discriminator": "entityComponentKind",
+      "properties": {
+        "entityComponentKind": {
```
Thank you for disambiguating the type, but I would offer a compromise here: normally, discriminators for the REST APIs might just be generic like "kind" here; but, to disambiguate in an SDK where multiple services / SDKs may have such a generic term, that's where you disambiguate. Thus, you'd set `x-ms-enum.name` to "entityComponentKind".

jinshang-msft committed:
[Hub Generated] Review request for Microsoft.Media/Encoding to add version stable/2022-07-01 (#21190)

* Adds base for updating Microsoft.Media/Encoding from version stable/2021-11-01 to version 2022-07-01

* Updates readme

* Updates API version in new specs and examples

* Copy over the new swagger and example files, and modify the readme file

* Fix the missing identifier properties

* Fix the experimentalOptions in StandardEncoderPreset

* Run Prettier fix

* Restore the original order of the existing models

* Revert the casing change in api version description

* fix spell check and missing format

* add subwoofer word back to custom words

* Add test scenarios

* fix missing ids in examples

* Revert "fix missing ids in examples"

This reverts commit b84e104dd86dab6c2ce796a8bccfa2efc8432997.

* Revert "Add test scenarios"

This reverts commit 9456029757435de2800698350528cf067f5bf490.

Co-authored-by: giakas <giakas@microsoft.com>

Wzb123456789 committed:
[monitor] Update readme.python.md (#22163)

* Update readme.python.md

* Update readme.python.md

* Update readme.python.md

* Update readme.python.md

* Update readme.md

Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>

weidongxu-microsoft committed:
mgmt, autorest configure for Java codegen (#22122)

dw511214992 committed:
fix cadl sample (#22146)

* fix

* Update Widgets_ListWidgetsSample.json

* Update Widgets_ListWidgetsSample.json

jinshang-msft committed:
[Hub Generated] Review request for Microsoft.Media/Encoding to add version preview/2022-05-01-preview (#21160)

* Adds base for updating Microsoft.Media/Encoding from version stable/2021-11-01 to version 2022-05-01-preview

* Updates readme

* Updates API version in new specs and examples

* Copy over the swagger files for 2022-05-01-preview

* Fix validation failures

* Fix validation failures

* Fix validation failures II

* Fix readme file

* Fix readme file II

* Fix readme file III

* Restore the original order of the existing models

* Revert the casing change in api version description

raych1 committed:
Added shared library guideline (#22162)

* Added shared library guidline

* Use sample project in example

pchoudhary-microsoft committed:
Sync private repo (#22074)

leal-alvaro committed:
The run.sh script requires git, which is not available by default (#22174)

* The run.sh script requires git, but git is not available by default in the base image

I've created a shell in the datacontainer and noticed that the run.sh script fails, because the git command is not found.  Without it, the datacontainer cannot check out the local clone of the repo and we end up with a completely empty/usr/data/openapispecs. This is currently blocking RPaaS api-validation service in Air-Gapped clouds, where we depend entirely on specs loaded from the data container.

I tested this change locally: WIth git being previously installed to the image, run.sh succeeds and /usr/data/openapispecs is properly populated with the repo contents.

* Addressing PR commants

Use Ubuntu 22.04, as suggested in the PR reviews.

gaoyp830 committed:
Dev reservations microsoft.capacity 2022 11 01 (#21565)

* Adds base for updating Microsoft.Capacity from version stable/2022-03-01 to version 2022-11-01

* Updates readme

* Updates API version in new specs and examples

* 2022-02-16-privatepreview CalculateExchange and Exchange, 2022-06-02-privatepreview Reservation properties change

* Add appliedScopeProperties to purchase and patch request. Add reviewDateTime to purchase patch request and get response for reservationOrder

* Add reviewDateTime to Reservation Response

* Update operationId to clean up doc heirarchy

* Modified examples to address property changes. Added back quota.json. Add readonly to property definitions

* Add 202 headers for Split, Merge, Patch. Add back AppliedScopes in reservation response to match request.

* Add nextlink and totaItems, filter, skip, take to GetCatalogs

* Fix lint error

* Add back publisherId, offerId, planId for GetCatalog API

* Revert "Update operationId to clean up doc heirarchy"

This reverts commit 31826f8e7ee40e30f099d5d95bf8b5fccf728819.

* Add back expiryDate and purchaseDate

Add back expiryDate and purchaseDate for backward compatibility

* Revert string to enum change

* Remove readOnly for existing definition to fix breaking change

* change expand to $expand

* Address comments from ARMReview

* Revert "Address comments from ARMReview"

This reverts commit cdeda23a6a666d79991b369bfc0aea31edd9d54e.

* Address comments for arm review

* Added x-ms-pageable for catalog api

* Fix spellcheck unknown words

* address comments and legacy lint errors and warnings

* add location for return. address some lint warnings. use proxyResource

* address legacy lint error and warnings

* Add descriptions for objects and fix typo

* fix spell error and missing description

* Fix comment

* Add back long running operations true to available scopes

* Add more description for appliedScopeProperties

Co-authored-by: chchen3 <chchen3@microsoft.com>

qianwens committed:
[service connector ]add default value for enums (#22099)

* add default value for enums

* add default value for enum

* remove

Alancere committed:
[Go] add network readme.go.md config (#22184)

AntonioVT committed:
[Hub Generated] Review request for Microsoft.ContainerRegistry to add version preview/2023-01-01-preview (#21727)

iVincentFeng committed:
[Hub Generated] Review request for Microsoft.StreamAnalytics to add version preview/2017-04-01-preview (#22090)

* Added the support for Delta Lake output for Azure Stream Analytics.

* Added missing references to Delta Lake samples

* Corrected the api versions in the examples.

* Resolved validation errors.

* Tried to fix LRO_RESPONSE_HEADER

* Tried to fix LRO_RESPONSE_HEADER

* Tried to fix LPO validation error.

* LPO

Co-authored-by: Vincent Feng <vifeng@microsoft.com>

iVincentFeng committed:
[Hub Generated] Review request for Microsoft.StreamAnalytics to add version preview/2021-10-01-preview (#22092)

* Added support for Delta Lake output

* Corrected api version

* Fixed LPO issues.

Co-authored-by: Vincent Feng <vifeng@microsoft.com>

leal-alvaro committed:
Fix wrong URL for Ubuntu (#22193)

The URL for Ubuntu 22.04 in MCR is slightly different from what we had before for Ubuntu 20.04. this is breaking the Docker build.

This PR replaces the wrong URL with the new and correct one.

enjaimes committed:
Dev search microsoft.search 2022 09 01 (#21964)

* Adds base for updating Microsoft.Search from version stable/2020-08-01 to version 2022-09-01

* Updates readme

* Updates API version in new specs and examples

* Add missing private endpoint connection properties (#21629)

* add missing private endpoint connection properties

* fix errors

* Adding delegated identities to data plane

* Revert "Adding delegated identities to data plane"

This reverts commit def1bd3ba296458c19b8f04cb1a7a5b800d002f9.

* Fix Lint ProvisioningStateValidation Error

* Applying prettier changes
npm run prettier -- --write

Co-authored-by: Enrique Jaimes <enjaimes@microsoft.com>

* Satumkur/new api ver 20220901 (#21922)

* semantic updates

* Add Dataplane RBAC related changes + examples

* Prettier fixes

* Fix LintDiff errors

* Update SearchUpdateServiceDisableLocalAuth.json

update erroneous api version param

Co-authored-by: Samartha Tumkur Vani <satumkur@microsoft.com>
Co-authored-by: Enrique Jaimes <enjaimes@microsoft.com>

* Add CMK Enforcement policy support (#21996)

* Add CMK Enforcement policy support

* Update api-version

* Remove unused properties

* Remove bypass property

Co-authored-by: satumkurmsft <72990508+satumkurmsft@users.noreply.github.com>
Co-authored-by: Matt <57731498+mattmsft@users.noreply.github.com>
Co-authored-by: Samartha Tumkur Vani <satumkur@microsoft.com>

saprakas committed:
[Hub Generated] Review request for Microsoft.ServiceFabricMesh to add version preview/2018-09-01-preview (#22171)

* Fixing Linter Validation Error

* Fixing Model Validation errors

* Incorporating review feedback.

sebastianrex committed:
add or modify files (#22187)

weidongxu-microsoft committed:
mgmt, java, fix invalid swagger (#22204)

AntonioHernandez16 committed:
fixing eTag issues (#22121)

cvarsh committed:
Recoveryservicesbackup microsoft.recovery services 2023-01-01 (#21788)

* Adds base for updating Microsoft.RecoveryServices from version stable/2022-10-01 to version 2023-01-01

* Updates readme

* Updates API version in new specs and examples

* Update bms.json

Added RecoveryPoint Expiry time for Immutability GA

* Update bms.json

Prettier fix

* Update readme.md

Updated default tag

* Update bms.json

Updated bms json for expiry time format..

* Update bms.json

UPdating correct type

* Update bms.json

Syntax error fixed

* Update bms.json

Prettier fix

paul-selvaraj committed:
Release maintenance microsoft.maintenance 2022 11 01 preview (#21806)

* Adds base for updating Microsoft.Maintenance from version preview/2022-07-01-preview to version 2022-11-01-preview

* Updates readme

* Updates API version in new specs and examples

* Add Override Functionality to the Maintenance Configuration.

* Adding Maintenance Override Properties to the Maintenance Configuration.

* Revert "Adding Maintenance Override Properties to the Maintenance Configuration."

This reverts commit 28d2f7556eb10d8aeb7fe28e4c27489fa5e6ae19.

* Adding Maintenance Override Properties to the Maintenance Configuration.

* Adding "x-ms-identifiers": [] to fix lint error.

* Adding description to the Override Properties.

* Swagger Prettier Check Fix

* Reverting the Changes in the 07-01 preview version

* Address PR comments

* Fixing the Datetime type

* Fixing Code Checks

* Fixing DateTime

* Removing Name from the json

* Triggering a API Check run

* Pretty Check

* Changed Description

Co-authored-by: Paul Selvaraj <paulse@microsoft.com>

Commit:
add or modify files (#22189)

Commit:
[LOADTESTSERVICE] added "format" to "Endpoint", removed "continuationToken" (#21986)

* added "format" to "Endpoint"

* Update loadtestservice.json

Removed `continuationToken` and `maxPageSize`

* re-added "maxpagesize"

Co-authored-by: Nivedit Jain <jain.22@iitj.ac.in>

ostojicb committed:
Bump InstanceFailoverGroups version (#22172)

rkmanda committed:
Fix operation status common type to bring it up to speed with RpaaS's runtime contract (#22192)

* Add resourceId as an optional property for operation status response schema

* base commit cloning the v4 type

* adding resourceId to the operation status response as an optional property to make the common type compliant with RpaaS's runtime behavior

* prettier

msyyc committed:
Update readme.python.md (#22221)

tomkerkhove committed:
feat: Introduce 11 new events for Azure API Management in Azure Event Grid system topic (#21771)

* feat: Introduce 11 new events for Azure API Management in Azure Event Grid system topic

Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

* Add CE samples

Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

* Run prettier

Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

* Rename

Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

* "API" -> "Api"

* Rename events

Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

* Add sample links

Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

aakash049 committed:
Added data plane swagger for Microsoft.DataBox RP for EventGrid system topic (#21945)

* Added data plane swagger for Microsoft.DataBox EventGrid system topic

* Update DataBox.json

* updated

* Update copy_completed.json

* Update copy_started.json

* Update order_completed.json

* Update DataBox.json

* Resolving comments around Stage details & enum

* resolving arch board points

* Update DataBox.json

jimmyca15 committed:
[Hub Generated] Publish private branch 'user/jimmyca/AppConfigurationV2' (#22213)

* Set foundation for api version 2.0

* Update to v2.

* API review updates.
* Update API version for 2.0 to use date format. 2.0 -> 2022-11-01-preview.
* Remove unnecessary if-match and if-none-match to snapshot creation.

* * Fixed prettier, oav, autorest validation errors.
* Added missing examples.
* Added readme tag for new api version.

* Addressed feedback.
* Add error schema.
* Creation should return 201 response.
* Remove application/json from consumes.

* Add descriptions. Remove breaking changes caused by retroactively adding x-ms-enum.

* * Remove default schema for head requests. They have no response body.
* Add requirements on snapshot filters.

* Add retention_period, status_code, and failed state..

* Fix snapshots path (snapshot/ -> snapshots/). Remove extraneous items_link property from snapshot.

* Feedback.

* Respect enum changes introduced in prior API version: https://github.com/Azure/azure-rest-api-specs/issues/21216

fix number format lint diff errors.

navba-MSFT committed:
{AzureApplicationInsights} fixes Azure/azure-rest-api-specs#22136 IgnoreHttpsStatusCode should be IgnoreHttpStatusCode (#22199)

fixes Azure/azure-rest-api-specs#22136 

IgnoreHttpsStatusCode should be IgnoreHttpStatusCode.

More Info here: https://learn.microsoft.com/en-us/rest/api/application-insights/web-tests/get?tabs=HTTP#webtest

john35452 committed:
[Maps - RevGeoCoding] Add version preview/2022-12-01-preview [Hub Generated] (#19951)

* Adds base for updating Search from version preview/2022-02-01-preview to version 2022-09-01-preview

* Updates readme

* Updates API version in new specs and examples

* 2022/07/26 Add new version swagger for RevGC

* 2022/07/27 Update description

* 2022/08/03 Add batch RevGC api

* 2022/08/04 Change batch operationId to Post

* 2022/08/04 Change pickType to matchedType

* 2022/08/04 Add preview warning and change latlng order

* 2022/08/05 Update operationId

* 2022/08/09 Change status code for partial failure in batch api

* 2022/08/09 Use array for matchedType

* 2022/08/10 Add optional id to the batch example

* 2022/08/10 Add enum for matchedType

* 2022/08/11 Change point to query

* 2022/08/22 Update matchedType to matchTypes

* 2022/08/22 Update batch api path

* 2022/08/22 Add geoJsonData and move nextLink

* 2022/08/22 Remove the description of the case of multiple entities are returned

* 2022/08/22 Add matchTypes in the example query

* 2022/08/22 Update error in batch api examples

* 2022/08/22 Update description for 207 batch response

* 2022/08/22 Change content-type to Json

* 2022/08/24 Update entity type in the example

* 2022/08/24 Remove depulicated types in the description

* Revert "2022/08/22 Add geoJsonData and move nextLink"

This reverts commit f2ffa7ba701f6fc73f7a8db31c8e88de20ce84b9.

* Revert "2022/08/22 Change content-type to Json"

This reverts commit 0af80bb25bd97c0b8f66669d529a19dfaa3a4050.

* 2022/09/27 Change MatchTypes to ResultTypes

* 2022/09/28 Change query to location in RevGC

* 2022/09/28 Add nextLink in batch response

* 2022/10/03 Add intersection in address

* 2022/10/03 Add type to the item of adminDistricts

* 2022/10/04 Update description for latlong pair

* 2022/10/04 Update GeoJson type

* 2022/10/04 Add query to forward batch input parameters

* 2022/10/04 Remove default api-version

* 2022/10/12 Change location to coordinates

* 2022/11/18 Add int type

* 2022/11/18 Add x-ms-enum for enums

* 2022/11/28 Remove except query and update url

* 2022/11/28 Update description for 207 response

* 2022/12/01 Update description for query

* 2022/12/08 Fix operationId

* 2023/01/05 Upgrade api-version to 2022-12-01-preview

* 2023/01/06 Copy swagger file and examples from 2022-09-01

* 2023/01/06 Update examples

* 2023/01/10 Update swagger file

* 2023/01/10 Update swagger file and examples

* 2023/01/10 Update readme.md

* 2023/01/11 Update resultTypes

* 2023/01/12 Update adminDistrict for RevGC

Co-authored-by: John Lai <tingchiehlai@microsoft.com>

AhmedLeithy committed:
Rename Language 2023-05-01 to 2023-04-01 (#21990)

* Rename Language 2023-05-01 to 2023-04-01

* Adding missing header to sample (#22242)

Co-authored-by: Kareem Yousef <kayousef@microsoft.com>
Co-authored-by: Kareem Yousef <121571242+kayousef@users.noreply.github.com>
Co-authored-by: Joel Hendrix <jhendrix@microsoft.com>

CordeliaGrey committed:
[Hub Generated] Publish private branch 'jiwang6/workflow-api' (#22241)

* Add purview workflow api.

* Fix prettier check errors

* Fix workflow tag

* Fix spell check error

* Fix linter error

* Fix spell error and remove redundant example

* Remove readme.md

* Fix linter warning

* Add reassign workflow request API

* Add nextLink in example file

* Fix spell error

* Fix example value

* fix opertaion id

* Fix some issue

* Fix reassign task request operationId

* Fix azure_auth scope

* Re-organize workflow API

* Add enum option for some parameters

* Fix enum option for some parameters

* Rename request type filter

* Add description for top level resource schema

* Add description for top level resource schema

* Add RequestEventPayloadWebRequest multiple sub models

* Rename and change the model of some properties

* Update workflow swagger

* Update workflow swagger for SDK board review

* Update workflow app model

* Remove list/get/cancel userrequest api

* lint error fix

* Fix lint error

* Fix small swagger issue

* fix small swagger issue

* Add unknow words

* fix small swagger issue

* Fix small swagger issue

* Fix some properties

* Fix swagger

Co-authored-by: Jing Wang <jiwang6@microsoft.com>

weidongxu-microsoft committed:
CODEOWNERS, add @Azure/api-stewardship-board to data-plane (#22183)

* add @Azure/api-stewardship-board to data-plane

* Update CODEOWNERS

Co-authored-by: Mike Kistler <mikekistler@microsoft.com>

jianyexi committed:
Update cadl validation doc (#22098)

* wip

* update ci-fix for cadl validation

* remove unused file

* update

* update cadl validation

Co-authored-by: jianyexi <jianyxi@microsoft.com>

divya-jay committed:
Rename MonitoringAccount to Monitor workspace in tags, operationId and description (#22182)

* Rename MonitoringAccount to Monitor workspace in tags, operationId and description

* Revert renaming of definitions

* revert updated property name

Co-authored-by: Divya Jayaraman <divyaj@microsoft.com>

kayousef commented:
Replaced with this PR https://github.com/kayousef/azure-rest-api-specs/pull/1

kayousef commented on dev/cognitiveservices/data-plane/Language/analyzetext.json:
```diff
@@ -849,7 +849,7 @@
       "allOf": [
         {
-          "$ref": "common.json#/definitions/PreBuiltTaskParameters"
+          "$ref": "#/definitions/CustomTaskParameters"
```
Not a regression. This is a new custom task I'm adding in the PR so it should be using `CustomTaskParameters` but I mistakenly used `PrebuiltTaskParameters` at first.
